### ISSUE TITLE
Matching of Users and Authors

### ISSUE THREAD

**Issue by bmaland** [hadley/crantastic#22]
I was thinking that it could be done similarly to the way Ohloh does it, e.g. that there is a "this is me" button visible on author pages for logged in users. When clicked, the author id would be linked with the user. This could potentially be abused, but any misbehaving users could be blocked (I don't really think it will be a problem though). Alternatively, there could be a second step where a link gets emailed to the Author mail address and the user has to click it (will be a hassle for author profiles with invalid email addresses though, e.g. "at" instead of "@").

**Comment by hadley** [comment]
That sounds reasonable.  Could we require the user to make put in some effort to reduce the potential for abuse?  e.g. http://recaptcha.net/ 

If it turns out to be a problem, we could require an extra field in the DESCRIPTION file like crantasticid: 103, 156

**Comment by bmaland** [comment]
Recaptcha protection is added in hadley@a15aaa1ff43c5add3bcaf37f29117c1e3fb13097

**Comment by hadley** [comment]
Is this pulic in the ui yet?

**Comment by hadley** [comment]
Actually it might need to be on the package page as well so that authors (and not just maintainers) can associate themselves with a package. This might need some db changes unfortunately

**Comment by bmaland** [comment]
The hard part is extracting the correct author names from the author fields -- there's a lot of prose etc. there. :) Getting closer, though, and the rest of the code is working well.

**Comment by hadley** [comment]
I thought you would just ignore that and allow the user to say that they're any author of any package. What ate you thinking?

**Comment by bmaland** [comment]
Yes but currently authors == maintainers (only maintainers are stored in the db as Authors). Also the current Version <-> Author relation only exists for maintainers as well, so there is no automatic way to detect if a non-maintainer author authored a given package.

An alternative route would be to give the users more control, e.g. giving them the power to claim that they are a package author (entering the details into a form, resulting in a new Author record) and then claim that they wrote a package through the user interface. This would then ignore any data from the fields in DESCRIPTION (or maybe just use some parts of it).

Would you prefer the latter, or maybe some combination?

**Comment by bmaland** [comment]
A related important issue is that the same author can have is named spelled slightly differently in different DESCRIPTION files. E.g. "Eun-kyung Lee" and "Eun-Kyung Lee" are now stored as two different authors even though it clearly is the same person. I propose to create a merging-function, available only to admins/moderators, that makes it possible to merge authors together. This can be done by keeping track of a list of aliased names, updating existing relations to point to the correct author id, and finally delete the old record.

**Comment by hadley** [comment]
Can we just not side step this issue entirely by allowing any user to say they are a contributor to a package?  This reificiation of author names is a tricky process.

**Comment by bmaland** [comment]
I'm starting to agree - depending on automatic recognition here is too error-prone (or would require large amounts of work).

It was suggested earlier to highlight "expert" reviews - maybe we could define users that has contributed to N packages as being experts? That would add some extra incentive for users to register their contributions.

**Comment by hadley** [comment]
Maybe we could have little tags next to user names? Maybe just the number of packages they have contributed to?  With a description title attribute so you could see that 3 actually meant "Has contributed to 3 packages"
